File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        vector<int> table(26, 0);
        int size = s.size();
        int left = 0;
        int max_appears = 0, max_window = 0;
        for (int right = 0; right < size; right++)
        {
            table[s[right] - 'A']++;

            max_appears = max(max_appears, table[s[right] - 'A']);
            while ((right - left + 1) - max_appears > k)
            {
                table[s[left] - 'A']--;
                left++;
            }

            max_window = max(max_window, right - left + 1);
        }
        return max_window;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def hei(head):
            if not head:
                return 0
            left = hei(head.left)
            if left == -1:
                return -1
            right = hei(head.right)
            if right == -1:
                return -1
            if abs(left - right) > 1:
                return -1
            return 1 + max(left, right)
        return hei(root) != -1